src/framework/components/collision/system.js

```diff
@@ -331,6 +333,29 @@ class CollisionMeshSystemImpl extends CollisionSystemImpl {
     // special handling
     beforeInitialize(component, data) {}
 
+    createAmmoHull(mesh, node, shape, scale) {
+        const hull = new Ammo.btConvexHullShape();
+
+        const point = new Ammo.btVector3();
+
+        const positions = [];
+        mesh.getPositions(positions);
+
+        for (let i = 0; i < positions.length; i += 3) {
+            point.setValue(positions[i] * scale.x, positions[i + 1] * scale.y, positions[i + 2] * scale.z);
+
+            // No need to calculate the aabb here. We'll do it after all points are added.
+            hull.addPoint(point, false);
+        }
+
+        Ammo.destroy(point);
+
+        hull.recalcLocalAabb();
+        hull.setMargin(0.01);   // Note: default margin is 0.04
+
+        shape.addChildShape(this.system._getNodeTransform(node), hull);
```

Comment:
I assume the hull will be destroyed when the shape is destroyed, no need to do it manually.

Reply:
Correct.